### PR TITLE
Changes needed for mediation debugger wire logs

### PR DESCRIPTION
**Pull request by lankavitharana**

### CODE REVIEW

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpClientConnection.java [hunk 1]
```diff
@@ -105,11 +108,83 @@ public void consumeInput(final NHttpClientEventHandler handler) {
         if (this.log.isDebugEnabled()) {
             this.log.debug(this.id + ": Consume input");
         }
-        super.consumeInput(handler);
-        if (isReleaseConn()) {
-            if (handler instanceof TargetHandler) {
-                ((TargetHandler) handler).getTargetConfiguration().getConnections().releaseConnection(this);
-                this.setReleaseConn(false);
+//        super.consumeInput(handler);
```
Comment (raviu) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpClientConnection.java]: Why have we removed calling super.consumeInput()? Have we brought that into the logger class? Is it possible to do what we need only here and call super.consumeInput(), reason being, if there are fixes to HTTP Core library at this level, we may have to manually fix those in here. Is there a perf reason for doing this?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpClientConnection.java [hunk 2]
```diff
@@ -119,7 +194,65 @@ public void produceOutput(final NHttpClientEventHandler handler) {
         if (this.log.isDebugEnabled()) {
             this.log.debug(this.id + ": Produce output");
         }
-        super.produceOutput(handler);
+//        super.produceOutput(handler);
```
Comment (raviu) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpClientConnection.java]: Same comment as consumeInput() above.

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpServerConnection.java [hunk 3]
```diff
@@ -103,15 +107,152 @@ public void consumeInput(final NHttpServerEventHandler handler) {
         if (this.log.isDebugEnabled()) {
             this.log.debug(this.id + ": Consume input");
         }
-        super.consumeInput(handler);
+//        super.consumeInput(handler);
```
Comment (raviu) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpServerConnection.java]: Same comment as for Client handler.

File: modules/core/src/main/java/org/apache/synapse/core/SynapseEnvironment.java [hunk 4]
```diff
@@ -263,7 +263,7 @@
      *
      * @return whether debugging is enabled in the environment
      */
-    public boolean isDebugEnabled();
+    public boolean isDebuggerEnabled(); //todo refactor this "isDebuggerEnabled"
```
Comment (chanakaudaya) [inline modules/core/src/main/java/org/apache/synapse/core/SynapseEnvironment.java]: This todo item is no longer needed since it has already been done.

**Comment by raviu** [comment]
This PR has been superseded by https://github.com/wso2/wso2-synapse/pull/530/

**Pull request closed without merging**